### PR TITLE
Migrate EmsRefresh.refresh queue args to data

### PR DESCRIPTION
**Pull request by agrare**
Move existing EmsRefresh.refresh queue args to data for performance.

Related to: https://github.com/ManageIQ/manageiq/pull/16271

### CODE REVIEW

**Comment by Fryguy** [comment]
As discussed, since Marshal.dump/load is not guaranteed to work across Ruby versions, we should bracket this around upgrades and just delete rows that can't be Marshaled (which if one fails they all will likely fail).

We may want to be resilient on the MiqQueue side as well...not sure we ever caught that error there.

**Comment by agrare** [comment]
@Fryguy added tests with msg_data containing an empty array with marshal format version `3.0`

File: spec/migrations/20171025122732_move_ems_refresh_args_to_data_spec.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    it "If there are no args" do
+      q_item = miq_queue_stub.create!(refresh_queue_options.merge(:msg_data => nil))
+
+      migrate
+
+      expect(q_item.reload.args).to match_array([])
+    end
+
+    it "Handle invalid marshal format errors" do
+      # "\x03\x00\x00" is an empty array in Marshal version 3.0
```
Comment (Fryguy) [inline spec/migrations/20171025122732_move_ems_refresh_args_to_data_spec.rb]: Is there a version of this for the up migration?

Comment (agrare) [reply spec/migrations/20171025122732_move_ems_refresh_args_to_data_spec.rb]: I wasn't sure how to test `Marshal.dump` failing due to a version issue, ideas?

Comment (agrare) [reply spec/migrations/20171025122732_move_ems_refresh_args_to_data_spec.rb]: Let me try putting an anonymous class in the args, that should do it.

Comment (agrare) [reply spec/migrations/20171025122732_move_ems_refresh_args_to_data_spec.rb]: Scratch that, that won't even serializing the value to args

Comment (Fryguy) [reply spec/migrations/20171025122732_move_ems_refresh_args_to_data_spec.rb]: Oh I see...it's only the load part that fails.  :+1:

**Pull request merged**